v-next/hardhat/src/internal/core/hook-manager.ts

```diff
@@ -205,6 +196,42 @@ export class HookManagerImplementation implements HookManager {
     );
   }
 
+  async #getHandlersInChainedRunningOrder<
+    HookCategoryNameT extends keyof HardhatHooks,
+    HookNameT extends keyof HardhatHooks[HookCategoryNameT],
+  >(
+    hookCategoryName: HookCategoryNameT,
+    hookName: HookNameT,
+  ): Promise<Array<HardhatHooks[HookCategoryNameT][HookNameT]>> {
+    const pluginHooks = await this.#getPluginHooks(hookCategoryName, hookName);
+
+    const dynamicHooks = await this.#getDynamicHooks(
+      hookCategoryName,
+      hookName,
+    );
+
+    return [...dynamicHooks, ...pluginHooks];
+  }
+
+  async #getHandlersInSequentialRunningOrder<
+    HookCategoryNameT extends keyof HardhatHooks,
+    HookNameT extends keyof HardhatHooks[HookCategoryNameT],
+  >(
+    hookCategoryName: HookCategoryNameT,
+    hookName: HookNameT,
+  ): Promise<Array<HardhatHooks[HookCategoryNameT][HookNameT]>> {
+    const reversedPluginHooks = (
+      await this.#getPluginHooks(hookCategoryName, hookName)
+    ).reverse();
+
+    const dynamicHooks = await this.#getDynamicHooks(
+      hookCategoryName,
+      hookName,
+    );
+
+    return [...dynamicHooks, ...reversedPluginHooks];
```

Comment:
```suggestion
    const reverseDynamicHooks = await this.#getDynamicHooks(
      hookCategoryName,
      hookName,
    ).reversed();

    return [...reversedPluginHooks, ...reversedDynamicHooks];
```

Shouldn't this be like this?

Reply:
I have applied the complete reverse approach:

```ts
return this.#getHandlersInChainedRunningOrder(hookCategoryName, hookName).reverse();
```

The tests and doc comments have been updated to reflect the complete reordering.

See [`3dd37fe` (#5751)](https://github.com/NomicFoundation/hardhat/pull/5751/commits/3dd37fe886712111f7ba183f94e6c9d97e73ccbd)